test/test_c10d.py

```diff
@@ -509,10 +515,103 @@ def test_gloo_backend(self):
     @skip_if_not_multigpu
     @skip_if_not_nccl
     def test_nccl_backend(self):
-        store = c10d.TCPStore('localhost', self.port, self.rank == 0)
+        store = c10d.TCPStore('localhost', self.port, self.is_master)
         process_group = c10d.ProcessGroupNCCL(store, self.rank, self.world_size)
         self._test_ddp_with_process_group(process_group)
 
+    @skip_if_not_multigpu
+    def test_dist_broadcast_coalesced(self):
+        # Set up process group.
+        store = c10d.TCPStore('localhost', self.port, self.is_master)
+        options = c10d.ProcessGroupGloo.Options()
+        options.devices = [c10d.ProcessGroupGloo.create_tcp_device(interface="lo")]
+        process_group = c10d.ProcessGroupGloo(store, self.rank, self.world_size, options)
+
+        device = torch.device('cuda')
+
+        target = torch.arange(10, dtype=torch.float64, device=device).chunk(2)
+
+        if self.is_master:
+            # All processes should have these tensors in the end.
+            tensors = target
+        else:
+            # Non-master processes start with empty tensors and should be
+            # filled with the tensors from the master.
+            tensors = torch.zeros(10, device=device).chunk(2)
+
+        c10d._dist_broadcast_coalesced(
+            tensors,
+            buffer_size=25 * MB,
+            process_group=process_group)
+
+        if not self.is_master:
+            self.assertEqual(tensors, target)
+
+    @skip_if_not_multigpu
+    def test_sync_params_no_buffers(self):
+        # Set up process group.
+        store = c10d.TCPStore('localhost', self.port, self.is_master)
+        options = c10d.ProcessGroupGloo.Options()
+        options.devices = [c10d.ProcessGroupGloo.create_tcp_device(interface="lo")]
+        process_group = c10d.ProcessGroupGloo(store, self.rank, self.world_size, options)
+
+        # Use all available devices on every process here (data is small, so should be fine).
+        devices = list(range(torch.cuda.device_count()))
```

Comment:
I am thinking of dividing the devices among all processes, wdyt?

Reply:
We do this in other tests, but in a lot of cases we only run with two GPUs, so with two processes that would only leave one device per process and make this test less meaningful. Is this reasonable?

Reply:
it should be fine for Gloo, but NCCL it won't work AFAIK. We can leave as is